sdk/core/Azure.Core/src/Pipeline/AzureOperationScope.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+
+namespace Azure.Core.Pipeline
+{
+    public struct DiagnosticScope: IDisposable
+    {
+        private Activity _activity;
+
+        private readonly string _name;
+
+        private readonly DiagnosticListener _source;
+
+        internal DiagnosticScope(string name, DiagnosticListener source)
+        {
+            _name = name;
+            _source = source;
+            _activity = _source.IsEnabled() ? new Activity(_name) : null;
+        }
+
+        public bool IsEnabled => _activity != null;
+
+        public void AddAttribute(string name, string value)
+        {
+            _activity?.AddTag(name, value);
+        }
+
+        public void AddAttribute<T>(string name, T value)
+        {
+            if (_activity != null)
+            {
+                AddAttribute<T>(name, value, t => t.ToString());
```

Comment:
Any reason not to call the `AddAttribute(name, value.ToString())` overload here instead?